Plant trial export download

Nuanda opened the pull request:
Fixes #683 #684 #690. Partially addresses #680 #639.

kammerer commented on spec/services/submission/plant_trial_exporter_spec.rb:
```diff
@@ -21,40 +26,32 @@
-        to match_array [ts2.score_value, '-']
-      expect(documents[:trait_scoring].lines[1,2].map{ |l| l.strip.split(',')[3] }).
-        to match_array [ts3.score_value, '-']
+      expect(generated_scores.map{ |l| l[0] }).to match_array [ts1.score_value, '-', '-']
```
I got the impression that order of scores should be important here, shouldn't it?

Nuanda commented:
@kammerer  Both comments addressed. Please go ahead with the review.

kammerer commented:
:+1:

Nuanda commented:
@kammerer Tomasz, I have added one more commit - to make the CSV headers consistent with the Browse section naming, and also removed two useless columns in one of the documents. Please see the commit and check if I should change anything.

Nuanda commented:
I'll go ahead and merge-deploy that since time is short and we want the client's feedback. Meanwhile I'll leave the branch so any further changes could be made on it.

Nuanda commented:
Note to self:

- [ ] test escaping commas in textual fields
- [ ] do something with design factors field export

Nuanda commented:
Moved all changes to #692
